modules/compatibility/src/test/java/org/apache/ignite/compatibility/sql/randomsql/RandomQuerySupplier.java

```diff
@@ -202,6 +205,15 @@ private Select rndSingleTableSelect(RandomisedQueryContext rndQryCtx) {
     private Ast rndQueryFilter(RandomisedQueryContext rndQryCtx) {
         List<TableRef> scopeTbls = rndQryCtx.scopeTables();
 
+        RandomisedQueryContext parentCtx = rndQryCtx.parentContext();
+        if (parentCtx != null) {
+            return new BiCondition(
+                pickRndItem(pickRndItem(scopeTbls).cols(), NUMERICAL_COLUMN_FILTER),
+                pickRndItem(pickRndItem(parentCtx.scopeTables()).cols(), NUMERICAL_COLUMN_FILTER),
+                Operator.EQUALS
+            );
+        }
+
         Ast cond1 = rndTableFilter(rndQryCtx, scopeTbls.get(rnd.nextInt(scopeTbls.size())));
```

Comment:
I know that it isn't scope of this task, but can we replace 
    scopeTbls.get(rnd.nextInt(scopeTbls.size()))) 
with 
    pickRndItem(scopeTbls) 
here to improve code readability?

Reply:
Done